hawkular/alerts.py

```diff
@@ -0,0 +1,159 @@
+"""
+   Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+   and other contributors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+"""
+
+from __future__ import unicode_literals
+import client
+
+try:
+    # Python 3
+    from urllib.parse import quote, urlencode
+except ImportError:
+    # Fall back to Python 2's urllib2
+    from urllib import quote, urlencode
+
+try:
+    import simplejson as json
+except ImportError:
+    import json
+
+
+class ApiJsonEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, ApiOject):
+            return obj.to_json_object()
+        else:
+            return json.JSONEncoder.default(self, obj)
+
+
+class ApiOject:
+    def to_api_object(self, dict):
+        pass
+
+    def to_json_object(self):
+        return self.transform_dict_to_camelcase(self.__dict__)
+
+    def _to_camelcase(self, word):
+        s = ''.join(x.capitalize() or '_' for x in word.split('_'))
+        return ''.join([s[0].lower(), s[1:]])
+
+    def _to_underscore(self, word):
+        return ''.join(["_" + c.lower() if c.isupper() else c for c in word]).strip('_')
+
+    def transform_dict_to_camelcase(self, dictionary):
+        return dict((self._to_camelcase(k), v) for k, v in dictionary.iteritems() if v is not None)
+
+    def transform_dict_to_underscore(self, dictionary):
+        return dict((self._to_underscore(k), v) for k, v in dictionary.iteritems() if v is not None)
+
+    @classmethod
+    def transform_to_object(cls, o):
```

Comment:
@rubenvp8510 since the need for these conversions in the API should be pretty clear, I think we should split it in explicit `dict_to_object(cls, o)` and `list_to_objects(cls, o)`. This will also allow us to drop all these `isinstance` (who are not aligning well with duck-typing in python) and have a consistent return type (an single object or a list of objects).